src/ScheduledJobs/Repositories/ParticipantRegistrationDynamoDbRepository.cs

```diff
@@ -24,14 +25,39 @@ public ParticipantRegistrationDynamoDbRepository(IAmazonDynamoDB client, IDynamo
             _config = new DynamoDBOperationConfig { OverrideTableName = awsSettings.ParticipantRegistrationDynamoDbTableName };
         }
 
-        public async Task<IEnumerable<Participant>> GetAllAsync()
+        public Task<IAsyncEnumerable<Participant>> GetAllAsync()
         {
-            return await _context.ScanAsync<Participant>(null, _config).GetRemainingAsync();
+            return Task.FromResult(GetAllAsyncEnumerable());
         }
 
+        public async IAsyncEnumerable<TModel> GetAllMappedAsync<TModel>(Func<Participant, TModel> mapper)
+        {
+            await foreach (var participant in GetAllAsyncEnumerable())
+            {
+                yield return mapper(participant);
+            }
+        }
+
+        private async IAsyncEnumerable<Participant> GetAllAsyncEnumerable()
+        {
+            var search = _context.ScanAsync<Participant>(null, _config);
```

Comment:
It looks like a synchronous `Scan` returns an `IEnumerable` and does the paging automatically behind the scenes. It might have better performance out of the box.

Reply:
@con-nihr can you send documentation for that?  Docs I was looking at say its only `ScanAsync` in .net core

Reply:
Yes, you're right not available in .net core